Source/WebKit/NetworkProcess/NetworkLoadChecker.cpp

```diff
@@ -254,12 +254,13 @@ void NetworkLoadChecker::checkRequest(ResourceRequest&& request, ContentSecurity
 {
     ResourceRequest originalRequest = request;
 
-    if (auto* contentSecurityPolicy = this->contentSecurityPolicy()) {
+    if (CheckedPtr contentSecurityPolicy = this->contentSecurityPolicy()) {
         if (this->isRedirected()) {
             auto type = m_options.mode == FetchOptions::Mode::Navigate ? ContentSecurityPolicy::InsecureRequestType::Navigation : ContentSecurityPolicy::InsecureRequestType::Load;
             contentSecurityPolicy->upgradeInsecureRequestIfNeeded(request, type);
         }
         if (!this->isAllowedByContentSecurityPolicy(request, client)) {
+            contentSecurityPolicy = nullptr;
```

Comment:
Because the call to the `handler` below may cause the ContentSecurityPolicy object to get destroyed, *before* the CheckedPtr goes out of scope. I found this out the hard way.